doc/examples/segmentation/plot_morphsnakes.py

```diff
@@ -91,28 +86,35 @@ def _store(x):
 
 ax[1].imshow(ls, cmap="gray")
 ax[1].set_axis_off()
-contour = ax[1].contour(evolution[2], [0.5], colors='g')
-contour.collections[0].set_label("Iteration 2")
-contour = ax[1].contour(evolution[7], [0.5], colors='y')
-contour.collections[0].set_label("Iteration 7")
-contour = ax[1].contour(evolution[-1], [0.5], colors='r')
-contour.collections[0].set_label("Iteration 35")
-ax[1].legend(loc="upper right")
+
+ax[1].contour(evolution[2], [0.5], colors='g')
+ax[1].contour(evolution[7], [0.5], colors='y')
+ax[1].contour(evolution[-1], [0.5], colors='r')
+
+# `Axes.legend` does not support handles for `QuadContourSet`, define proxy
+# artists instead
+contour_labels = [
+    mlines.Line2D([], [], color="g", label="Iteration 2"),
+    mlines.Line2D([], [], color="y", label="Iteration 7"),
+    mlines.Line2D([], [], color="r", label="Iteration 35"),
+]
```

Comment:
Can we pop this into a for-loop, and simplify the comment above contour_labels? I think the current text will be a bit confusing to newcomers, one of our target audiences. Perhaps something like "placeholder labels for contours" or similar.

Reply:
(I can also push the change, let me know.)

Reply:
There is also the [legend_elements](https://matplotlib.org/stable/api/contour_api.html#matplotlib.contour.ContourSet.legend_elements) method, which essentially creates proxy artists for you.

Reply:
Thanks @rcomer; I tried that, but I couldn't figure out how to make it work in our case, since `set_label` doesn't seem to affect the returned label elements. Is there a way to set the label on each contour that way, so it comes back via `legend_elements`?